hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -511,143 +514,145 @@ static inline size_t sizeofLargestProgramSection(uint64_t addr, size_t size)
 {
-    if ((amountLeftToOperate == PROGRAM_PHRASE_SIZEOF_INLINE_DATA) ||
-        ((currentOperatingStorageAddress % SIZEOF_DOUBLE_PHRASE) == PROGRAM_PHRASE_SIZEOF_INLINE_DATA)) {
-        setup8ByteWrite(currentOperatingStorageAddress, currentOperatingData);
+    if ((context->amountLeftToOperate == PROGRAM_PHRASE_SIZEOF_INLINE_DATA) ||
+        ((context->currentOperatingStorageAddress % SIZEOF_DOUBLE_PHRASE) == PROGRAM_PHRASE_SIZEOF_INLINE_DATA)) {
+        setup8ByteWrite(context->currentOperatingStorageAddress, context->currentOperatingData);
 
-        amountLeftToOperate            -= PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
-        currentOperatingStorageAddress += PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
-        currentOperatingData           += PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
+        context->amountLeftToOperate            -= PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
+        context->currentOperatingStorageAddress += PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
+        context->currentOperatingData           += PROGRAM_PHRASE_SIZEOF_INLINE_DATA;
     } else {
-        size_t amount = sizeofLargestProgramSection(currentOperatingStorageAddress, amountLeftToOperate);
-        setupProgramSection(currentOperatingStorageAddress, currentOperatingData, amount);
+        size_t amount = sizeofLargestProgramSection(context->currentOperatingStorageAddress, context->amountLeftToOperate);
+        setupProgramSection(context->currentOperatingStorageAddress, context->currentOperatingData, amount);
 
-        amountLeftToOperate            -= amount;
-        currentOperatingStorageAddress += amount;
-        currentOperatingData           += amount;
+        context->amountLeftToOperate            -= amount;
+        context->currentOperatingStorageAddress += amount;
+        context->currentOperatingData           += amount;
     }
 }
 
 /**
  * Advance the state machine for erase. This function is called only if
  * amountLeftToOperate is non-zero.
  */
-static inline void setupNextErase(void)
+static inline void setupNextErase(struct mtd_k64f_data *context)
 {
-    setupEraseSector(currentOperatingStorageAddress);     /* Program FCCOB to load the required command parameters. */
+    setupEraseSector(context->currentOperatingStorageAddress); /* Program FCCOB to load the required command parameters. */
 
-    amountLeftToOperate            -= ERASE_UNIT;
-    currentOperatingStorageAddress += ERASE_UNIT;
+    context->amountLeftToOperate            -= ERASE_UNIT;
+    context->currentOperatingStorageAddress += ERASE_UNIT;
 }
 
-static int32_t executeCommand(void)
+static int32_t executeCommand(struct mtd_k64f_data *context)
 {
     launchCommand();
 
     /* At this point, The FTFE reads the command code and performs a series of
      * parameter checks and protection checks, if applicable, which are unique
      * to each command. */
 
-    if (asyncOperationsEnabled()) {
-        /* Asynchronous operation */
+#if ASYNC_OPS
```

Comment:
> Please explain if you have a strong objection. Otherwise let's avoid loosing time over minor disagreements.

+1

> Not a fan of 14a14a0 but otherwise LGTM

Please provide more details why not a fan and how it could be.